mlflow/pyfunc/model.py

```diff
@@ -73,16 +74,57 @@ def _log_warning_if_params_not_in_predict_signature(logger, params):
         )
 
 
-class PythonModel:
+class _PythonModelMetaclass(ABCMeta):
+    def __call__(cls, *args, **kwargs):
+        cls._warn_on_setting_model_in_init()
+        return super().__call__(*args, **kwargs)
+
+    def _warn_on_setting_model_in_init(cls):
+        if not cls.__bases__ == (PythonModel,):
+            return
+
+        import click
+
+        message = (
+            click.style(
+                "It looks like you're trying to save a model as an instance attribute. ",
+                fg="yellow",
+            )
+            + click.style("This is not recommended ", fg="yellow", bold=True)
+            + click.style(
+                "as it can cause problems with model serialization, "
+                "especially for large models. ",
+                fg="yellow",
+            )
+            + click.style("Please use the `artifacts` parameter", fg="yellow", bold=True)
+            + click.style(
+                ", and load your external model in the `load_context()` method instead.",
+                fg="yellow",
+            )
+        )
+
+        try:
+            model_param_present = "model" in inspect.signature(cls.__init__).parameters
+            model_assignment = (
+                cls.load_context == PythonModel.load_context
+                and "self.model =" in inspect.getsource(cls.__init__)
+            )
```

Comment:
/ai I want to check if the code contains assignment that looks like `self.model = model`. Write me a function for that using the ast module.

Reply:
This should be more precise.

Reply:
i was thinking of being able to catch stuff like 
```
import huge_model

class MyModel(mlflow.pyfunc.PythonModel):
    def __init__(self):
        self.model = huge_model
```
do you think it's something users might do?

Reply:
I have definitely seen people TRY to do that, but that introduces pickling errors for sure. This is one of the patterns that we'd like to catch at object instantiation time (before trying to save to MLflow)